src/mode/terminal.rs

```diff
@@ -41,6 +42,30 @@ use crate::{
 use core::{cmp::min, fmt};
 use hal::{blocking::delay::DelayMs, digital::v2::OutputPin};
 
+pub trait TerminalDisplaySize: DisplaySize {
+    type CharNum: Unsigned;
```

Comment:
Similar to my `DisplaySize` comment, this could be:

```suggestion
    const CHARNUM: u8;
```